platform-sdk/swirlds-platform-core/src/testFixtures/java/com/swirlds/platform/test/fixtures/addressbook/RandomRosterEntryBuilder.java

```diff
@@ -0,0 +1,195 @@
+/*
+ * Copyright (C) 2024 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.swirlds.platform.test.fixtures.addressbook;
+
+import static com.swirlds.common.test.fixtures.RandomUtils.randomIp;
+
+import com.hedera.hapi.node.base.ServiceEndpoint;
+import com.hedera.hapi.node.state.roster.RosterEntry;
+import com.hedera.pbj.runtime.io.buffer.Bytes;
+import com.swirlds.common.platform.NodeId;
+import com.swirlds.platform.crypto.SerializableX509Certificate;
+import com.swirlds.platform.test.fixtures.crypto.PreGeneratedX509Certs;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.security.cert.CertificateEncodingException;
+import java.util.Objects;
+import java.util.Random;
+
+/**
+ * A builder for creating random {@link RosterEntry} instances.
+ */
+public class RandomRosterEntryBuilder {
+
+    private final Random random;
+    private NodeId nodeId;
+    private Long weight;
+    private Integer port;
+    private String hostname;
+    private SerializableX509Certificate sigCert;
+
+    private long minimumWeight = 0;
+    private long maximumWeight = Long.MAX_VALUE / 1024;
+
+    /**
+     * Creates a new {@link RandomRosterEntryBuilder} instance.
+     *
+     * @param random the random number generator to use
+     * @return a new {@link RandomRosterEntryBuilder} instance
+     */
+    @NonNull
+    public static RandomRosterEntryBuilder create(@NonNull final Random random) {
+        return new RandomRosterEntryBuilder(random);
+    }
+
+    /**
+     * Constructor.
+     *
+     * @param random the random number generator to use
+     */
+    private RandomRosterEntryBuilder(@NonNull final Random random) {
+        this.random = Objects.requireNonNull(random);
+    }
+
+    /**
+     * Builds a new {@link RosterEntry} instance.
+     *
+     * @return a new {@link RosterEntry} instance
+     */
+    @NonNull
+    public RosterEntry build() {
+
+        // Future work: use randotron utility methods once randotron changes merge
```

Comment:
Presumably this comment is left over from the old roster generation test code, since randotron has merged a while ago.

Reply:
Yes, this class is a refactored copy of the `RandomAddressBuilder`. I removed this comment from here (the AddressBuilder will eventually be removed completely, so I'm not touching it.)